src/pages/index.js

```diff
@@ -1,211 +1,269 @@
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "298fcaaa-e173-4e81-9326-ea95e21a4625",
+    "Content-Type": "application/json",
+  },
+});
+
+// Instantiate user info
+const user = new UserInfo({
+  title: ".profile__title",
+  subtitle: ".profile__subtitle",
+  avatar: ".profile__image",
+});
+
+// Instantiate card section
+export const cardSection = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (item) => {
-      section.addItem(createCard(item));
+      const cardElement = createCard(
+        item,
+        handleCardImageClick,
+        handleDeleteClick,
+        handleLikeClick
+      );
+      cardSection.addItem(cardElement);
     },
   },
   "#card-list"
 );
-section.renderItems();
 
-// User Info
-const user = new UserInfo({
-  title: ".profile__title",
-  subtitle: ".profile__subtitle",
-});
+// ! INITIAL DATA FETCHING
+
+// Fetch initial cards and user info
+const initialize = async () => {
+  try {
+    const [cards, userInfo] = await Promise.all([
+      api.getInitialCards(),
+      api.getUserInfo(),
+    ]);
+
+    cardSection.setItems(cards);
+    user.setUserInfo({
+      name: userInfo.name,
+      about: userInfo.about,
+    });
+    user.setUserAvatar({
+      avatar: userInfo.avatar,
+    });
+  } catch (err) {
+    console.error("Initialization error:", err);
+  }
+};
+
+initialize();
 
-// ! EVENT LISTENERS FOR POPUPS
+// ! EVENT HANDLERS
+
+// Handle profile edit form submission
+function handleProfileEditSubmit(profileData) {
+  return handleSubmit(() => {
+    return api
+      .updateProfileInfo({
+        name: profileData.title,
+        description: profileData.subtitle,
+      })
+      .then((updatedData) => {
+        user.setUserInfo(updatedData);
+      });
+  }, profileEditPopup);
+}
 
+// Add card handler
+function handleAddCardSubmit(inputValues) {
+  return handleSubmit(() => {
+    return api
+      .addCard({ name: inputValues.title, link: inputValues.image })
+      .then((newCardData) => {
+        const newCardElement = createCard(
+          newCardData,
+          handleCardImageClick,
+          handleDeleteClick,
+          handleLikeClick
+        );
+        cardSection.addItem(newCardElement);
+      });
+  }, addCardPopup);
+}
+
+// Handle profile image form submission
+function handleProfileImageSubmit(newImageData) {
+  return handleSubmit(() => {
+    return api
+      .updateProfileAvatar({ avatar: newImageData.image })
+      .then((res) => {
+        user.setUserAvatar({ avatar: res.avatar });
+      });
+  }, profileImagePopup);
+}
+
+// Handle card delete confirmation
+function handleDeleteClick(card) {
+  if (!card._cardElement) {
+    console.error("Card element is null for card:", card);
+  }
+
+  cardToDelete = card;
+  deleteCardPopup.open();
+}
+
+// Handle card like button
+function handleLikeClick(card) {
+  const newLikeState = !card.isLiked;
+  card.updateLikeState(newLikeState);
```

Comment:
```js

  card.updateLikeState(newLikeState);
  card.isLiked = newLikeState;
```

You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.



```js
// Handle card like button
function handleLikeClick(card) {
  const newLikeState = !card.isLiked;

  if (newLikeState) {
    api
      .likeCard(card.getId())
      .then(() => {
        card.updateLikeState(newLikeState);
        card.isLiked = newLikeState;
      })
      .catch((err) => console.error(err));
  } else {
    api
      .unlikeCard(card.getId())
      .then(() => {
        card.updateLikeState(newLikeState);
        card.isLiked = newLikeState;
      })
      .catch((err) => console.error(err));
  }
}

```